stores/hostdb.go

```diff
@@ -426,6 +499,155 @@ func (ss *SQLStore) Host(ctx context.Context, hostKey types.PublicKey) (hostdb.H
 	}, nil
 }
 
+func (ss *SQLStore) HostInfo(ctx context.Context, autopilotID string, hk types.PublicKey) (hi api.HostInfo, err error) {
+	err = ss.db.Transaction(func(tx *gorm.DB) error {
+		var entity dbHostInfo
+		if err := tx.
+			Model(&dbHostInfo{}).
+			Where("db_autopilot_id = (?)", gorm.Expr("SELECT id FROM autopilots WHERE identifier = ?", autopilotID)).
+			Where("db_host_id = (?)", gorm.Expr("SELECT id FROM hosts WHERE public_key = ?", publicKey(hk))).
+			Preload("DBHost").
+			First(&entity).
+			Error; errors.Is(err, gorm.ErrRecordNotFound) {
+			if err := tx.
+				Model(&dbAutopilot{}).
+				Where("identifier = ?", autopilotID).
+				First(nil).
+				Error; errors.Is(err, gorm.ErrRecordNotFound) {
+				return api.ErrAutopilotNotFound
+			} else if err := tx.
+				Model(&dbHost{}).
+				Where("public_key = ?", publicKey(hk)).
+				First(nil).
+				Error; errors.Is(err, gorm.ErrRecordNotFound) {
+				return api.ErrHostNotFound
+			}
+			return api.ErrHostInfoNotFound
+		} else if err != nil {
+			return err
+		}
+		hi = entity.convert()
+		return nil
+	})
+	return
+}
+
+func (ss *SQLStore) HostInfos(ctx context.Context, autopilotID string, filterMode, usabilityMode, addressContains string, keyIn []types.PublicKey, offset, limit int) (his []api.HostInfo, err error) {
+	if offset < 0 {
+		return nil, ErrNegativeOffset
+	}
+
+	err = ss.db.Transaction(func(tx *gorm.DB) error {
+		// fetch ap id
+		var apID uint
+		if err := tx.
+			Model(&dbAutopilot{}).
+			Where("identifier = ?", autopilotID).
+			Select("id").
+			Take(&apID).
+			Error; errors.Is(err, gorm.ErrRecordNotFound) {
+			return api.ErrAutopilotNotFound
```

Comment:
This makes for very awkward UX. We should just leave the additional information blank. Otherwise the UI might try to fetch 50 hosts to display, the whole thing fails because a single host doesn't have any information on it and then the UI needs to go through that list one-by-one.

Reply:
I'm not sure I understand what you are referring to here. This line essentially returns "404 autopilot not found" when the user requests `/api/bus/autopilot/doesntexists/hosts`. Did you mean to comment on the line where I do something similar for the hosts not being found?